ddl/db_integration_test.go

```diff
@@ -1946,3 +1948,53 @@ func (s *testIntegrationSuite3) TestParserIssue284(c *C) {
 	tk.MustExec("drop table test.t_parser_issue_284")
 	tk.MustExec("drop table test.t_parser_issue_284_2")
 }
+
+func (s *testIntegrationSuite6) TestAddExpressionIndex(c *C) {
```

Comment:
Adding more test cases:
1. Create a expression index and then create a normal column has same name with its hidden column name.
2. Create a normal column, and then create a expression index, its hidden column name are same with prev column.
3. After above cases, drop expression index,  and then drop column. 
4. In case 3, you can do this as alternative: drop normal column. and then drop expression index.